gui-js/libs/shared/src/lib/constants/constants.ts

```diff
@@ -180,6 +181,30 @@ export const commandsMapping = {
   CANVAS_ITEM_TOGGLE_PLOT_TAB_DISPLAY:
     '/minsky/canvas/item/togglePlotTabDisplay',
 
+  CANVAS_RAVEL_NEXT_REDUCTION: '/minsky/canvas/item/nextReduction',
+  CANVAS_RAVEL_TOGGLE_CALIPERS: '/minsky/canvas/item/toggleDisplayFilterCaliper',
+  CANVAS_RAVEL_BROADCAST_TO_LOCKGROUP: '/minsky/canvas/item/broadcastStateToLockGroup',
+  CANVAS_RAVEL_HANDLE_SET_REDUCTION: '/minsky/canvas/item/handleSetReduction',
+  CANVAS_RAVEL_HANDLE_SELECTED: '/minsky/canvas/item/selectedHandle',
+  CANVAS_RAVEL_HANDLE_GET_SORT_ORDER: '/minsky/canvas/item/sortOrder',
+  CANVAS_RAVEL_HANDLE_SET_SORT_ORDER: '/minsky/canvas/item/setSortOrder',
+  CANVAS_RAVEL_HANDLE_SET_SORT_BY_VALUE: '/minsky/canvas/item/sortByValue',
+  CANVAS_RAVEL_HANDLE_SET_DIMENSION: '/minsky/canvas/item/setDimension',
+  CANVAS_RAVEL_HANDLE_DIMENSION_TYPE: '/minsky/canvas/item/dimensionType',
+  CANVAS_RAVEL_HANDLE_DIMENSION_UNITS: '/minsky/canvas/item/dimensionUnitsFormat',
+  CANVAS_RAVEL_HANDLE_DESCRIPTION: '/minsky/canvas/item/handleDescription',
+  CANVAS_RAVEL_HANDLE_SET_DESCRIPTION: '/minsky/canvas/item/setHandleDescription',
+  CANVAS_RAVEL_HANDLE_ALL_SLICE_LABELS: '/minsky/canvas/item/allSliceLabels',
+  CANVAS_RAVEL_HANDLE_PICKED_SLICE_LABELS: '/minsky/canvas/item/pickedSliceLabels',
+  CANVAS_RAVEL_HANDLE_PICK_SLICE_LABELS: '/minsky/canvas/item/pickSliceLabels',
+  CANVAS_RAVEL_LOCKGROUP: '/minsky/canvas/item/lockGroup',
+  CANVAS_RAVEL_LOCKGROUP_HANDLELOCKINFO: '/minsky/canvas/item/lockGroup/handleLockInfo',
+  CANVAS_RAVEL_LOCKGROUP_ALL_LOCK_HANDLES: '/minsky/canvas/item/lockGroup/allLockHandles',
+  CANVAS_RAVEL_LOCK_RAVELS_IN_SELECTION: '/minsky/canvas/lockRavelsInSelection',
+  CANVAS_RAVEL_LOCKGROUP_SET_LOCK_HANDLES: '/minsky/canvas/item/lockGroup/setLockHandles',
+  CANVAS_RAVEL_LOCKGROUP_RAVELNAMES: '/minsky/canvas/item/lockGroup/ravelNames',
+  CANVAS_RAVEL_LOCKGROUP_VALIDATE_LOCK_HANDLE_INFO: '/minsky/canvas/item/lockGroup/validateLockHandleInfo/',
+
```

Comment:
Yeah - personally I think this commandsMapping structure is a mistake. I'd prefer to have the string explicitly coded at the call site, not buried away in this enum in a totally different part of the codebase, as it adds one level of abstraction too many. But this can be fixed up with a future refactor.

I'm working on a wrapper that turned these REST calls into standard Typescript calls, with proper compile time type checking, so that'll get rid of most of this nonsense :).

Reply:
Sounds good. Definitely in line with my thinking, as we've discussed before.